maat-court-data-api/src/main/java/gov/uk/courtdata/eform/repository/entity/EformsHistory.java

```diff
@@ -0,0 +1,37 @@
+package gov.uk.courtdata.eform.repository.entity;
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hibernate.annotations.CreationTimestamp;
+
+import java.time.LocalDateTime;
+
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+@Entity
+@Table(name = "EFORMS_HISTORY", schema = "TOGDATA")
+public class EformsHistory {
+    @Id
+    @Column(name = "ID")
+    @SequenceGenerator(name = "eforms_history_seq", sequenceName = "S_GENERAL_SEQUENCE", allocationSize = 1, schema = "TOGDATA")
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "eforms_history_seq")
+    private Integer id;
+    @Column(name = "USN", nullable = false)
+    private Integer usn;
+    @Column(name = "REP_ID")
+    private Integer repId;
+    @Column(name = "ACTION")
+    private String action;
+    @Column(name = "KEY_ID")
+    private Integer keyId;
+    @CreationTimestamp
+    @Column(name = "DATE_CREATED", nullable = false)
+    private LocalDateTime dateCreated;
+    @Column(name = "USER_CREATED", nullable = false)
```

Comment:
A minor consideration, in the DB layer the USER_CREATED field is indeed `NOT NULL` although it also has `DEFAULT user` defined.
I'm not sure if we want to enforce `nullable = false` at the API layer, maybe ensuring that the field is populated more accurately than the default? Or, maybe we should have it as `nullable` in the API layer and let the DB populate it with the default if it is not supplied.

Reply:
@stone-dj - Removed nullable for USER_CREATED field